### ISSUE TITLE
Error compiling stand code in R on 64-bit Windows machine

### ISSUE THREAD

**Issue by ddemleinad** [stan-dev/stan#2201]
#### Summary:
Please provide a short couple sentence summary.

I have a Windows 7 64-bit laptop and I am trying to install stan. In this respect I followed the instructions here;

https://github.com/stan-dev/rstan/wiki/Installing-RStan-on-Windows

However when I run some code I get the error;

Error in compileCode(f, code, language = language, verbose = verbose) : 
  Compilation ERROR, function(s)/method(s) not created! file2fa058e9722f.cpp:1:0: sorry, unimplemented: 64-bit mode not compiled in
 
which I assume indicates there is some kind of issue with the C++ compiler in Rtools - i.e. I installed the 32-bit version (Rtools34.exe in fact) rather than the 64-bit version. 

What is my best course of action now - can I simply install the 64-bit Rtools on top of the 32-bit version or do I need to unintall the 32-bit version first? I am also not sure which 64-bit Rtools version to install from this webpage;

https://cran.r-project.org/bin/windows/Rtools/

Any help would be greatly appreciated!

**Comment by bgoodri** [comment]
What do you have in

readLines(normalizePath("~/.R/Makevars"))

If you google for the phrase "sorry, unimplemented: 64-bit mode not
compiled in", there is a bunch of hits.

**Comment by ddemleinad** [comment]
Thanks bgoodri for this - after running your code I get;


> readLines(normalizePath("~/.R/Makevars"))
[1] ""                                                                                  
[2] "CXXFLAGS=-O3 -mtune=native -march=native -Wno-unused-variable -Wno-unused-function"
[3] ""                                                                                  
[4] "CXXFLAGS += -Wno-ignored-attributes -Wno-deprecated-declarations"      

I will indeed take a look at your recomended Google search.

**Comment by ddemleinad** [comment]
It seems my installation of R tools did not install gcc at al but rather something called MinGW. When I run;

Sys.getenv('PATH')

I get ;

C:\\Program Files\\R\\R-3.2.2\\bin\\x64;c:\\Rtools\\bin;c:\\Rtools\\mingw_32\\bin; ....

So I see the path does not point to the 64-bit version of minggw. When I changed it to point to;

`C:\\Program Files\\R\\R-3.2.2\\bin\\x64;c:\\Rtools\\bin;c:\\Rtools\\mingw_64\\bin;
`
I overcome the 64-bit compile error I alluded to previoulsy but now get a whole load of more errors (sorry I know nothing about C++ code!). For example running this example C++ code;

```
library(inline) 
library(Rcpp)
src <- '
+ std::vector<std::string> s; 
+ s.push_back("hello");
+ s.push_back("world");
+ return Rcpp::wrap(s);
+ '
hellofun <- cxxfunction(body = src, includes = '', plugin = 'Rcpp', verbose = FALSE)
cat(hellofun(), '\n') 
hello world
```

I get;

```
Error in compileCode(f, code, language = language, verbose = verbose) : 
  Compilation ERROR, function(s)/method(s) not created! filee1c75b254fd.cpp:8:39: fatal error: stan/model/model_header.hpp: No such file or directory
 #include <stan/model/model_header.hpp>
                                       ^
compilation terminated.
make: *** [filee1c75b254fd.o] Error 1
Warning message:
running command 'make -f "C:/PROGRA~1/R/R-32~1.2/etc/x64/Makeconf" -f "C:/PROGRA~1/R/R-32~1.2/share/make/winshlib.mk" -f "//slhhome001/home$/u)55527/.R/Makevars" SHLIB_LDFLAGS='$(SHLIB_CXXLDFLAGS)' SHLIB_LD='$(SHLIB_CXXLD)' SHLIB="filee1c75b254fd.dll" WIN=64 TCLBIN=64 OBJECTS="filee1c75b254fd.o"' had status 2 


Any ideas?

```

**Comment by bgoodri** [comment]
The later error is caused by an incomplete or incorrect installation of the
StanHeaders package. So start a fresh R session without importing any old
stanfit objects and do

remove.packages("StanHeaders")

Then make sure both of these return ""

system.file("lib32", package = "StanHeaders")
system.file("lib64", package = "StanHeaders")

If not, you may have to manually remove the zombie StanHeaders directory.
Then do

install.packages("StanHeaders")
library(rstan)

**Comment by ddemleinad** [comment]
Many thanks for this - I will try this out as soon as I can.

**Comment by ddemleinad** [comment]
I have tracked the problem down to an unusual directory structure on my laptop that caused some sort of problem with the installation of stan and/or Rtools - i.e. my work laptop has strange directory paths like \\myhome001\home$\u)55527\stats (non project) presumably pointing to a server myhome001. This is somehow mapped locally to "MyDocuments", the intention being to duplicate any files onto the server when I am on the company network as a backup. R tried to install here and strugled to copy the downloaded packages from a temporary location to this strange directory path.

Anyway when I did a fresh install of everything but this time forcing R to install packages to a standard directory path then the whole thing worked.

Sorry for the confusion and thank-you so much to all those who took the time to help!
